cmd/gke-gcloud-auth-plugin/cred/gcloud_token_provider.go

```diff
@@ -0,0 +1,78 @@
+package cred
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// gcloudConfiguration holds types unmarshaled from gcloud config in json format
+type gcloudConfiguration struct {
+	Credential struct {
+		AccessToken string    `json:"access_token"`
+		TokenExpiry time.Time `json:"token_expiry"`
+	} `json:"credential"`
+	Configuration struct {
+		Properties struct {
+			Auth struct {
+				AuthorizationTokenFile string `json:"authorization_token_file"`
+			} `json:"auth"`
+		} `json:"properties"`
+	} `json:"configuration"`
+}
+
+//
+// gcloudTokenProvider provides gcloud Oath 2.0 tokens.
+//
+type gcloudTokenProvider struct {
+	readGcloudConfigRaw func() ([]byte, error)
+	readFile            func(filename string) ([]byte, error)
+}
+
+// readGcloudConfig returns an object which represents gcloud config output
+func (p *gcloudTokenProvider) readGcloudConfig() (*gcloudConfiguration, error) {
+	gcloudConfigbytes, err := p.readGcloudConfigRaw()
```

Comment:
nit: gcloudConfigBytes, but honestly could just do `b` or `bs`, maybe.

Reply:
Will correct casing (not new code, so will leave naming as is)